### PR TITLE
Query

### PR DESCRIPTION
**Pull request by memcmp**

### CODE REVIEW

File: src/main/java/org/springframework/data/crate/query/execution/QueryExecution.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            result = doExecute(query, values);
+        } catch (Exception e) {
+            return null;
+        }
+
+        if (result == null) {
+            return null;
+        }
+
+        // check if return type is the same with query returned object type
```
Comment (kovrus) [inline src/main/java/org/springframework/data/crate/query/execution/QueryExecution.java]: remove this comment, this check is not needed

File: README.md [hunk 2]
```diff
@@ -75,6 +75,37 @@ public interface UserRepository extends CrateRepository<User, String> {
 Currently documents can be queried by Id. Support will be added to derive
 queries from method names.
 
+#### Query Methods
+
+##### Query Lookup Strategies
+
+###### Declared Queries
+
+It's also possible to use the @Query annotation to define queries:
```
Comment (chaudum) [inline README.md]: `@Query`

File: README.md [hunk 3]
```diff
@@ -75,6 +75,37 @@ public interface UserRepository extends CrateRepository<User, String> {
...
+```java
+public interface UserRepository extends CrateRepository<User, String> {
+
+    @Query("select * from users")
+    List<User> getAllUsers();
+}
+```
+
+The query annotation can also be used with
+[Crate Parameter Substitution](https://crate.io/docs/reference/sql/rest.html#parameter-substitution),
```
Comment (chaudum) [inline README.md]: Crate Parameter Substitution -> parameter substitution

File: README.md [hunk 4]
```diff
@@ -75,6 +75,37 @@ public interface UserRepository extends CrateRepository<User, String> {
...
+where the arguments passed to the method has to match the positionals in the query:
```
Comment (chaudum) [inline README.md]: positionals -> positional placeholders

File: src/main/java/org/springframework/data/crate/query/execution/QueryExecution.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+    QueryExecution(CrateOperations operations) {
+        this.operations = operations;
+    }
+
+    public Object execute(CrateRepositoryQuery query, Object[] values) {
+        Assert.notNull(query);
+        Assert.notNull(values);
+
+        Object result;
```
Comment (chaudum) [inline src/main/java/org/springframework/data/crate/query/execution/QueryExecution.java]:
``` java
Object result = null;
try {
    result = doExecute(query, values);
} catch (Exception e) {
    // log something?
} finally {
    return result;
}
```

Comment (chaudum) [inline src/main/java/org/springframework/data/crate/query/execution/QueryExecution.java]: or:

``` java
try {
    return doExecute(query, values);
} catch (Exception e) {
    // log something?
}
return null;
```

File: src/main/java/org/springframework/data/crate/query/execution/QueryExecution.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+            return null;
+        }
+
+        if (result == null) {
+            return null;
+        }
+        return result;
+    }
+
+    protected abstract Object doExecute(CrateRepositoryQuery query, Object[] values);
```
Comment (chaudum) [inline src/main/java/org/springframework/data/crate/query/execution/QueryExecution.java:52]: `... throws Exception`?
maybe more specific?

File: src/test/java/org/springframework/data/crate/query/SimpleQueryCrateHandlerTest.java [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+public class SimpleQueryCrateHandlerTest {
+
+    private static CrateActionResponseHandler handler;
+
+    @BeforeClass
+    public static void beforeClass() throws NoSuchMethodException {
```
Comment (chaudum) [inline src/test/java/org/springframework/data/crate/query/SimpleQueryCrateHandlerTest.java]: why would it throw `NoSuchMethodException`?

**Comment by chaudum** [comment]
lgtm

**Pull request merged**